subsys/bluetooth/mesh/shell/shell_model_utils.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <bluetooth/mesh/models.h>
+#include "mesh/net.h"
+#include "mesh/access.h"
+
+struct bt_mesh_model *shell_model_first_get(uint16_t id)
+{
+	const struct bt_mesh_comp *comp = bt_mesh_comp_get();
+	struct bt_mesh_elem *elem;
+	struct bt_mesh_model *mod;
+
+	for (size_t i = 0; i < comp->elem_count; i++) {
+		elem = bt_mesh_elem_find(comp->elem[i].addr);
+		mod = bt_mesh_model_find(elem, id);
+		if (mod) {
+			return mod;
+		}
+	}
+
+	return NULL;
+}
+
+void shell_model_instances_get(uint16_t id, uint16_t *arr, uint8_t len)
+{
+	const struct bt_mesh_comp *comp = bt_mesh_comp_get();
+	struct bt_mesh_elem *elem;
+
+	for (size_t i = 0; i < len; i++) {
+		elem = bt_mesh_elem_find(comp->elem[i].addr);
+		if (bt_mesh_model_find(elem, id)) {
+			arr[i] = comp->elem[i].addr;
+		}
+	}
+}
+
+struct bt_mesh_model *shell_model_get(uint16_t id, uint16_t addr)
+{
+	struct bt_mesh_elem *elem = bt_mesh_elem_find(addr);
+
+	if (elem) {
+		return bt_mesh_model_find(elem, id);
+	}
+
+	return NULL;
+}
```

Comment:
Why does this also have some commands that gives the results from this functions? We can then give in any if and get the addresses printed, or something. Or we can send in a addr and get which id it has.

Reply:
I'm not sure if I understand. Can you elaborate?

Reply:
Doing a "mesh-util" command with a mod_id as input, prints the addr of all the models with that mod_id. This could also be done by giving a mod_id = ALL (solved by having no parameter), gives the users all mod_id and addr. This is just usefull functions for knowning which addresses different mod_id have a and a total overview over the different models the user have. This is just a suggestion.

Reply:
Ahh, ok I see. The first part is now available after the recent push. A generic version that gives you the full map of the elements and models would be nice to, but the question is if that is a shell command that should be present among the generic mesh functions. Otherwise we would potentially end up with multiple duplicates of the same shell feature, one for each model client that is added. Since this is a feature without dependencies on the rest of this task I think it is safe to postpone it for now.